pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/domain.go

```diff
@@ -55,10 +55,11 @@ func GenerateDomainName(name string, obj metav1.ObjectMeta, ingressConfig *v1bet
 		return "", fmt.Errorf("error rendering the domain template: %w", err)
 	}
 
-	urlErrs := validation.IsFullyQualifiedDomainName(field.NewPath("url"), buf.String())
-	if urlErrs != nil {
-		return "", fmt.Errorf("invalid domain name %q: %w", buf.String(), urlErrs.ToAggregate())
+	if !ingressConfig.DisableIngressCreation {
```

Comment:
I do not think this should be disabled, when ingress is turned off you still have an internal in cluster service url, you do not want that to exceed the dns limit.

Reply:
Isn't it just for generate a domain?
Internal services dont use domains, right? 

With that said, it would make sense to enable only when the ingress is being exposed, no?